ansible/roles/operator-pipeline/templates/openshift/tasks/merge-pr.yml

```diff
@@ -84,21 +80,27 @@ spec:
 
         # Do not merge if we do not have approval from the bot or any other repo member
         if ! grep "^APPROVED MEMBER " /tmp/reviews.txt ; then
-            echo "Skipping merge: PR is not approved."
+            echo -n "Skipping merge: PR is not approved." | tee "$(workspaces.output.path)/merge_error.txt"
             echo -n "false" > "$(results.pr_merged.path)"
             exit 0
         fi
 
         # Squash and merge only if the head commit sha has not changed since
         # the start of the pipeline run
-        gh pr merge "$(params.git_pr_url)" --squash --auto \
-            --match-head-commit "$(params.git_head_commit)"
+        MERGE_STDERR=$(gh pr merge "$(params.git_pr_url)" --squash --auto --match-head-commit "$(params.git_head_commit)" 2>&1 >/dev/null)
+        MERGE_RESULT=$?
+        MERGE_ERROR=$(echo "$MERGE_STDERR" | grep -oP '(?<=GraphQL: ).*')
 
-        if [[ $? -eq 0 ]] ; then
+        if [[ $MERGE_RESULT -eq 0 ]] ; then
             echo "PR has been merged!"
             echo -n "true" > "$(results.pr_merged.path)"
         else
             echo "Cannot merge PR"
             echo -n "false" > "$(results.pr_merged.path)"
-            exit 1
+            # Pass details about merge error to summary comment task
+            echo -n $MERGE_ERROR | tee "$(workspaces.output.path)/merge_error.txt"
+            # Fail pipeline if the error is not explicitly allowed.
+            if [[ "$MERGE_ERROR" != "Pull Request is still a draft (mergePullRequest)" && "$MERGE_ERROR" != "Head branch was modified. Review and try the merge again. (mergePullRequest)" ]] ; then
+              exit 1
+            fi
```

Comment:
The message "Head branch was modified. Review and try the merge again. (mergePullRequest)" may be a bit too obscure for the average end user. Since we are handling it as a special case anyway, I would replace the message shown to the user with something more clear like "The PR was not merged because another pipeline is already in progress: please wait for the result of the new tests" or something like that.